README.md

```diff
@@ -1,38 +1,89 @@
+### Components
+
+On top of this technical foundation, Argo provides a set of components that extensions can render. In remote-ui, the components your extension renders are actually just tiny JavaScript objects — they don’t have any DOM attached to them at all. These minimal components are sent to the host application to be rendered into native UI. This allows us to expose a minimal API for components that is focused on what you as a developer actually need to do — responding to events or customizing the appearance — while giving Shopify the ability to offer a tightly-optimized, highly-performant set of native UI components in the host application. Because your extension does not render to the actual DOM, Shopify can change or update the components without you needing to take any action, and we can make sure the components fit in seamlessly if they are in a context that allows merchant theming, like Shopify’s Checkout.
+
+The exact components available to the extension depend on the surface area you are embedding — please refer to the [documentation linked above](#i-just-want-to-build-an-argo-extension-not-learn-about-them) for references on what components you can use. If you’ve ever used [Polaris](https://polaris.shopify.com/), though, the components you will find will look extremely familiar, as many components and props are based on their Polaris equivalents.
+
+### Extension points
+
+While remote-ui provides the component model, and takes care of propagating updates to an extension’s UI into the host application, it does not have any built-in notion of an “extension” at all. To create an extension system, Shopify authors a bit of extra code: a mapping of “extension points” (which, in code, are just strings with a specific naming format), and a way to load third party code that can register to be called for those extension points. This might sound little complicated, but the code to put it all together is actually quite straightforward. The Argo ”runtime” code, which is used to construct the sandbox your extension will run in, looks almost exactly like this for all Argo-capable surface areas in Shopify:
+
+```js
+// We keep a mapping of all the extension points you register for
+const registeredExtensionPoints = new Map();
+
+// We define a globally-available `shopify` object. The only thing this object
+// can do is register an extension (which is just a callback function) for a
+// specific extension point.
+globalThis.shopify = {
+  extend(extensionPoint, extension) {
+    registeredExtensionPoints.set(extensionPoint, extension);
+  },
+};
+
+// Here’s the function the host application will call to load your extension’s
+// script into the sandbox
+export function load(script) {
+  // Internal workings — don’t worry too much about this, because we might change
+  // it in the future as new web platform features are available!
+}
+
+// Once your script is loaded, it has had its chance to register for extension
```

Comment:
```suggestion
// Once your script is loaded, it can register callbacks for extension points.
// Executing this function results in calling your registered callbacks with arguments from
// the host application, like the data your extension has access to and the UI “root“
// your components will be attached to.
```